### ISSUE TITLE
Mark a future task instance to success including downstream tasks using CLI

### ISSUE THREAD

**Issue by afsalabdup** [apache/airflow#9327]
Assume, there is a DAG with ID a_dag and five tasks with IDs dummy1, dummy2, dummy3, dummy4, dummy5. We want to mark the dummy3 task and all it's downstream tasks - dummy4,dummy5 etc  to success from the next DAG run. Since it's a future task instance, we can't do that in the UI. So we can use make use of CLI.

If we want to mark a future task instance to success, we can use below command.
airflow run -fAIim a_dag dummy3 2020-07-11T20:00:00+00:00

It would be great if we include downstream tasks in the above command, so that we do not have to run the above command for each downstream tasks separately.

**Comment by eladkal** [comment]
For anyone who wants to pick it up
this is an enhancment to [airflow tasks state](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#state_repeat1) i guess it should have similar flags like the [`clear`](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#clear)  command

**Comment by bhavaniravi** [comment]
I would like to work on this

**Comment by bhavaniravi** [comment]
- What is a future task instance? Something that's queued?
- @eladkal [airflow tasks state](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#state_repeat1) is currently read-only. Will this change give it write capabilities?

**Comment by afsalabdup** [comment]
Hi Bhavani,

This issue was opened by me. Future task instance can be any instance that
will run in the future. For example, if I want to put next week instance of
one of the tasks in a daily dag on hold.

Regards,
Afsal

**Comment by afsalabdup** [comment]
Thanks Bhavani for the explanation.

 I think we can mark a future dag to success using execution Id. I mean run
cli command to mark whole dag to success and the moment that particular
schedule is hit, we will see all tasks in success state in UI. I think have
tested it and worked fine, please correct me if I am wrong.
Can we implement similar feature for tasks ?

At present, if I want to skip a task then someone should be available or
monitor closely the scheduled run to mark that task to success. This is
risk as job might start before we update it's state, since we can see that
instance only once that particular schedule is hit.

Regards,
Afsal

**Comment by afsalabdup** [comment]
I am referring this -  "A dag run that hasn't even started yet, but is yet
to run in a future schedule."

Regards,
Afsal

**Comment by vdolez-happn** [comment]
Here a future DAG_run represents a DAG_run with an execution_date not yet reached by the scheduler (i.e now < execution_date + dag_period)

Here's a *Use Case* for this feature : 

A have a Sensor on a DAG which waits for an file.
I know that I won't receive that file tomorrow, and I want to ignore the sensor status for tomorrow's run.
I can backfill and mark_success the sensor, but this will create a DAG run for tomorrow which will be either `running` / `failed` / `success`. 

I don't want the DAG run to be in a finished state, but I don't want it to be running yet.

What I did then is I deleted the DAG run using the UI, and the DAG run successfully the following day.

**As an** Airflow user,
**I want to** change the state of a task in a DAG_run which will be trigger in the future by the scheduler without impacting other tasks scheduled for this specific DAG_run
**In order to** anticipate task_instance state and manually by-pass domain rules

**Comment by eladkal** [comment]
@bhavaniravi are you still working on this issue?

**Comment by iamluan** [comment]
Hi, is there any update on this issue? It's been open for quite a while, so I wanted to know if there has been any progress or if any help is needed. 
By the way, it seems that currently there is no CLI command that supports marking a task and all its downstream tasks of a not-yet-created DAG instance in the future as successful, and also the `airflow run` command is no longer available in the latest version.

**Comment by potiuk** [comment]
> Hi, is there any update on this issue? It's been open for quite a while, so I wanted to know if there has been any progress or if any help is needed.

If there is no update - it likely means there are no updaes. But you are free to take it over and propose PRs. Yes some commands were removed (see the release notes) - and there are likely other releated issues - but if you want I can assign you to make an investigation there and propose PRs or conclude that no fix is needed @iamluan

**Comment by iamluan** [comment]
Thanks @potiuk for replying. I think I should look for other issues.

**Comment by potiuk** [comment]
@bhavaniravi We are unassigning you from this issue as part of our updated [assignment policy](https://github.com/apache/airflow/blob/main/contributing-docs/04_how_to_contribute.rst#contribute-code-changes).

This is not meant to discourage your contribution — quite the opposite! You are still very welcome to work on this issue and submit a PR for it. Simply comment that you are working on it and open a PR when ready.

We found that formal assignments were not working well, as they often prevented others from contributing when the assignee was not actively working on the issue.
